supervisor/api/network.py

```diff
@@ -31,7 +31,7 @@
         vol.Optional(ATTR_ADDRESS): vol.Coerce(str),
         vol.Optional(ATTR_METHOD): vol.In(ATTR_METHODS),
         vol.Optional(ATTR_GATEWAY): vol.Coerce(str),
-        vol.Optional(ATTR_DNS): vol.Coerce(str),
+        vol.Optional(ATTR_DNS): vol.Schema([vol.Coerce(str)]),
```

Comment:
```suggestion
        vol.Optional(ATTR_DNS): [str],
```

I used the `Coerce` bad for string because it converts null to "None". If we need a string and they need not converted (because it could be a number), let's use only `str`